src/components/booking/ExpandedSessionCard.tsx

```diff
@@ -1,83 +1,66 @@
-/**
- * @author Angela Guo <aguo921@aucklanduni.ac.nz>
- */
-
-import { useState } from "react";
-import { IoCheckmarkCircle } from "react-icons/io5";
+import React, { useState } from "react";
 
+import { weekday } from "@/types/types";
 import { Card } from "../Card";
 import { LevelSelector } from "./LevelSelector";
 
 interface ExpandedSessionCardProps {
-  id: string;
+  day: weekday;
   startTime: Date;
   endTime: Date;
   location: string;
   address: string;
-  level?: string;
-  defaultLevel?: string;
-  setLevel: (level: string | undefined) => void;
 }
 
-const weekday = [
-  "Sunday",
-  "Monday",
-  "Tuesday",
-  "Wednesday",
-  "Thursday",
-  "Friday",
-  "Saturday",
-];
-
 export const ExpandedSessionCard = ({
+  day,
   startTime,
   endTime,
   location,
   address,
-  level,
-  setLevel,
-  defaultLevel,
-}: Omit<ExpandedSessionCardProps, "id">) => {
-  const [isOpen, setIsOpen] = useState(false);
-  const [pendingLevel, setPendingLevel] = useState<string | undefined>(level);
+}: ExpandedSessionCardProps) => {
+  const [showModal, setShowModal] = useState(false);
+  const [pendingLevel, setPendingLevel] = useState<string | undefined>(
+    undefined,
+  );
 
-  const dayOfWeek = weekday[startTime.getDay()];
+  const handleButtonClick = () => {
+    setShowModal(true);
+  };
+
+  const handleCloseModal = () => {
+    setShowModal(false);
+    setPendingLevel(pendingLevel);
+  };
 
   return (
     <Card className="relative font-normal">
```

Comment:
cut unnecessary classnames
```suggestion
    <Card>
```